### ISSUE TITLE
Rails/SaveBang doesn't detect when method return value is passed up the stack

### ISSUE THREAD

**Issue by RKushnir** [bbatsov/rubocop#3436]
## Expected behavior
Cop should detect and succeed when return value of `create`, `update` or `save` is checked up in the stack.

## Actual behavior
Cop tells to use a bang-version because the return value is not checked.

## Steps to reproduce the problem
Add an entry to `.rubocop.yml` to enable the cop
```yaml
Rails:
  Enabled: true

Rails/SaveBang:
  Enabled: true
```
Create a file with the following content
```ruby
def foo
  user.save
end

puts 'success' if foo
```
Run rubocop to get `Rails/SaveBang: Use save! instead of save if the return value is not checked.`

## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by arianf** [comment]
Rails/SaveBang is disabled by default

Add this into your `.rubocop.yml`

```yaml
Rails/SaveBang:
  Enabled: true
```

**Comment by RKushnir** [comment]
Thanks @arianf, I updated the issue to include that step :)

**Comment by bbatsov** [comment]
Yep, this shouldn't generate an offence if this is the final method invocation in a method body.

**Comment by denys281** [comment]
@RKushnir  Couldn't reproduce. 

Created file with: 

```
# file.rb
def foo
  user.save
end

puts 'success' if foo
```

Added to `.rubocop.yml`

```
Rails/SaveBang:
  Enabled: true
```

Result:

```
bundle exec bin/rubocop  file.rb                                                                            896ms  Wed Sep  7 23:31:17 2016
Inspecting 1 file
.

1 file inspected, no offenses detected

```

`0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15`

Did I missed something?

**Comment by RKushnir** [comment]
@denys281 No idea. Here's all the details I can provide:
```
➜  ruby -v
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]
➜  uname -a                  
Linux 4.2.0-35-generic #40-Ubuntu SMP Tue Mar 15 22:15:45 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
➜  ls -a 
.  ..  file.rb  Gemfile  Gemfile.lock  .rubocop.yml

### Gemfile
source "https://rubygems.org"
gem "rubocop"

### Gemfile.lock 
GEM
  remote: https://rubygems.org/
  specs:
    ast (2.3.0)
    parser (2.3.1.2)
      ast (~> 2.2)
    powerpack (0.1.1)
    rainbow (2.1.0)
    rubocop (0.42.0)
      parser (>= 2.3.1.1, < 3.0)
      powerpack (~> 0.1)
      rainbow (>= 1.99.1, < 3.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (~> 1.0, >= 1.0.1)
    ruby-progressbar (1.8.1)
    unicode-display_width (1.1.1)

PLATFORMS
  ruby

DEPENDENCIES
  rubocop

BUNDLED WITH
   1.11.2

### file.rb 
def foo
  user.save
end

puts 'success' if foo

### .rubocop.yml 
Rails/SaveBang:
  Enabled: true
```

**Comment by RKushnir** [comment]
@denys281 Right! You also need to enable the Rails cops with:
```yaml
Rails:
  Enabled: true
```

As I was running this inside a Rails project, I just assumed it's enabled.

**Comment by denys281** [comment]
@RKushnir  thank you!
